File: app/src/main/java/com/project/scratchstudio/kith_andoid/network/LiveDataHelper.java
package com.project.scratchstudio.kith_andoid.network;

import com.project.scratchstudio.kith_andoid.network.model.board.Board;
import com.project.scratchstudio.kith_andoid.network.model.comment.Comment;

import java.util.ArrayList;
import java.util.List;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MediatorLiveData;

public class LiveDataHelper {
    private MediatorLiveData<List<Comment>> commentList = new MediatorLiveData<>();
    private MediatorLiveData<List<Board>> boardList = new MediatorLiveData<>();
    private static LiveDataHelper liveDataHelper;

    public LiveDataHelper() {
    }

    synchronized public static LiveDataHelper getInstance() {
        if (liveDataHelper == null) {
            liveDataHelper = new LiveDataHelper();
        }
        return liveDataHelper;
    }

    public void updateCommentList(List<Comment> newCommentList) {
        List<Comment> list = new ArrayList<>();
        if (commentList.getValue() != null) {
            list.addAll(commentList.getValue());
        }
        list.addAll(newCommentList);
        commentList.postValue(list);
    }

    public LiveData<List<Comment>> observeCommentList() {
        return commentList;
    }

    public void clearCommentList() {
        if (commentList.getValue() != null) {
            commentList.getValue().clear();
        }
    }

    public void updateBoardList(List<Board> newBoardList) {
        List<Board> list = new ArrayList<>();
        if (boardList.getValue() != null) {
            list.addAll(boardList.getValue());
        }
        list.addAll(newBoardList);
        boardList.postValue(list);
    }

    public LiveData<List<Board>> observeBoardList() {
        return boardList;
    }
}

File: app/src/main/java/com/project/scratchstudio/kith_andoid/network/LiveDataHelper.kt
package com.project.scratchstudio.kith_andoid.network

import com.project.scratchstudio.kith_andoid.network.model.board.Board
import com.project.scratchstudio.kith_andoid.network.model.comment.Comment
import com.project.scratchstudio.kith_andoid.network.model.user.User

import java.util.ArrayList

import androidx.lifecycle.LiveData
import androidx.lifecycle.MediatorLiveData

class LiveDataHelper {
    private val userList = MediatorLiveData<List<User>>()
    private val commentList = MediatorLiveData<List<Comment>>()
    private val boardList = MediatorLiveData<List<Board>>()

    fun updateUserList(newUserList: List<User>) {
        val list = ArrayList<User>()
        if (userList.value != null) {
            list.addAll(userList.value!!)
        }
        list.addAll(newUserList)
        userList.postValue(list)
    }

    fun observeUserList(): LiveData<List<User>> {
        return userList
    }

    fun updateCommentList(newCommentList: List<Comment>) {
        val list = ArrayList<Comment>()
        if (commentList.value != null) {
            list.addAll(commentList.value!!)
        }
        list.addAll(newCommentList)
        commentList.postValue(list)
    }

    fun observeCommentList(): LiveData<List<Comment>> {
        return commentList
    }

    fun clearCommentList() {
        if (commentList.value != null) {
            commentList.value!!.clear()
        }
    }

    fun updateBoardList(newBoardList: List<Board>) {
        val list = ArrayList<Board>()
        if (boardList.value != null) {
            list.addAll(boardList.value!!)
        }
        list.addAll(newBoardList)
        boardList.postValue(list)
    }

    fun observeBoardList(): LiveData<List<Board>> {
        return boardList
    }

    companion object {
        private var liveDataHelper: LiveDataHelper? = null

        val instance: LiveDataHelper
            @Synchronized get() {
                if (liveDataHelper == null) {
                    liveDataHelper = LiveDataHelper()
                }
                return liveDataHelper!!
            }
    }
}
